rubix-core/src/main/java/com/qubole/rubix/core/CachedReadRequestChain.java

```diff
@@ -91,6 +91,7 @@ public Integer call()
                 directBuffer.get(readRequest.getDestBuffer(), readRequest.getDestBufferOffset() + nread, transferBytes);
                 leftToRead -= transferBytes;
                 nread += transferBytes;
+                directBuffer.clear();
```

Comment:
I'm not sure why this was required; flip should alone work. Without this getting some Out of bounds exceptions. Since this gets executed frequently it might add a considerable overhead. Will revisit this later